### DISCUSSION TITLE
Namespaces support

### DISCUSSION THREAD

**Discussion by fpliger** [pyscript/pyscript#1166] [Proposals]
As a user, I'd like Executable components (like <py-script>, <py-repl>, etc.. ) to be able to have/share a namespace in order to provide minimal isolation. This way, I have multiple components on the page that are actually not executing in the same scope.

his doesn't mean they are not sharing the same runtime, but rather that the execution scope is different (pyodide provides this in the form of dicts)

**Comment by JeffersGlass** [comment]
For those who find this discussion and are interested, it's probably worth looking at the back-and-forth in PR #503 , the first attempt at NameSpace support.
